iothub/device/src/Authentication/ClientAuthenticationForEdgeHubOnBehalfOf.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Authentication
+{
+    /// <summary>
+    /// Authentication method that generates shared access signature (SAS) token with refresh, based on a provided shared access key (SAK).
+    /// Build for gateway scenarios when one client authenticates on behalf of the other only.
```

Comment:
Can we mention in the comments that this only works for $edgeHub ?

`gateway scenarios` can be confusing, I think we mean "when using the $edgeHub in IoTEdge to authenticate on behalf of leaf devices or modules"